dis_snek/models/discord_objects/scheduled_event.py

```diff
@@ -0,0 +1,204 @@
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Union
+
+import attr
+from attr.converters import optional
+from dis_snek.const import MISSING
+from dis_snek.models.discord import DiscordObject
+
+from dis_snek.models.snowflake import Snowflake_Type, to_snowflake
+from dis_snek.models.timestamp import Timestamp
+from dis_snek.utils.attr_utils import define
+from dis_snek.utils.converters import timestamp_converter
+
+
+from enum import IntEnum
+
+if TYPE_CHECKING:
+    from dis_snek.client import Snake
+    from dis_snek.models.discord_objects.channel import GuildStageVoice, GuildVoice
+    from dis_snek.models.discord_objects.guild import Guild
+    from dis_snek.models.discord_objects.user import Member
+    from dis_snek.models.discord_objects.channel import TYPE_ALL_CHANNEL
+    from dis_snek.models.discord_objects.user import User
+    from dis_snek.models.snowflake import Snowflake_Type
+
+
+class ScheduledEventPrivacyLevel(IntEnum):
+    """The privacy level of the scheduled event."""
+
+    # Discord only has one at the momment.
+    GUILD_ONLY = 2
+
+
+class ScheduledEventEntityType(IntEnum):
+    """The type of entity that the scheduled event is attached to."""
+
+    STAGE_INSTANCE = 1
+    """ Stage Channel """
+    VOICE = 2
+    """ Voice Channel """
+    EXTERNAL = 3
+    """ External URL """
+
+
+class ScheduledEventStatus(IntEnum):
+    """The status of the scheduled event."""
+
+    # Discord only has one at the momment.
+    SCHEDULED = 1
+    ACTIVE = 2
+    COMPLETED = 3
+    CANCELED = 4
+
+
+@define()
+class ScheduledEventUser(DiscordObject):
+    user: "User" = attr.ib()
+    member: "Member" = attr.ib(default=MISSING)
```

Comment:
Why are you seperating user and member? 
`ScheduledEvent` is short for `GuildScheduledEvent`, they always occur within a guild so this *should* always be a member.

Reply:
Also member contains a user object

Reply:
I just saw that member could not be there sometimes 
https://discord.com/developers/docs/resources/guild-scheduled-event#get-guild-scheduled-event-users
If you could help with this one that would be nice, thanks.

Reply:
![image](https://user-images.githubusercontent.com/58155937/146618582-d8d1f259-9052-4293-953b-0bd3a5e256ff.png)

Reply:
https://discord.com/developers/docs/resources/guild-scheduled-event#guild-scheduled-event-user-object

`user` and `member` should remain separate as per the API schema.

Reply:
So is my method good? Or still need to be changed?

Reply:
I'd keep it as is

Reply:
I disagree with the above. One this object shouldn't even exist.

And two, Events ought to have an attribute called `members` or `participants` or whatever name which is a `List[Union[User, Member] `

That'd be the solution with the best UX

Reply:
> I disagree with the above. One this object shouldn't even exist.
> 
> And two, Events ought to have an attribute called `members` or `participants` or whatever name which is a `List[Union[User, Member] `
> 
> That'd be the solution with the best UX

I think the attributes existence should be respected if you're trying to keep it strict to the schema design, since a majority of every other object I've thus far seen does. From a UX perspective you're right, but from a developer transitioning from one library to another, I'm worried it might add on to confusion.

Reply:
We're not strictly following the discord dev schema, UX takes priority imo. I follow the docs where it doesnt harm UX

Reply:
Alright, I dig the thought. My only suggestion is maybe considering a property that can access that part of the API schema without actually having an attribute dedicated to it, so you get the best of both worlds and you don't end up *actually* writing a value for an attr you have to handle.

Reply:
I don't follow. How will a property provide access to the api-style schema?

Reply:
Not sure how you do it with your serializer mixin but something like this perhaps

```py
@property
def member(self):
    _member: "Member"
    return _member = attr.ib()
```

Reply:
I don't see what benifit that would offer? Nor do I really get what problem that code would solve?

Reply:
I ended up rewriting the code so that this object wasn't needed